src/integration/java/io/pinecone/integration/controlPlane/pod/CollectionTest.java

```diff
@@ -20,65 +20,81 @@
+    private static PineconeControlPlaneClient controlPlaneClient;
+    private static CollectionModel collection;
 
     @BeforeAll
     public static void setUp() throws InterruptedException {
         controlPlaneClient = new PineconeControlPlaneClient(apiKey);
-        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().pods(1).podType("p1.x1").replicas(1).environment(environment);
+
+        // Create and upsert to index
+        CreateIndexRequestSpecPod podSpec =
+                new CreateIndexRequestSpecPod().pods(1).podType("p1.x1").replicas(1).environment(environment);
         CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(podSpec);
-        PineconeConnection dataPlaneConnection = createNewIndexAndConnect(controlPlaneClient, indexName, dimension, indexMetric, spec);
+        PineconeConnection dataPlaneConnection = createNewIndexAndConnect(controlPlaneClient, indexName, dimension,
+                indexMetric, spec);
         VectorServiceGrpc.VectorServiceBlockingStub blockingStub = dataPlaneConnection.getBlockingStub();
         indexes.add(indexName);
 
-        blockingStub.upsert(buildRequiredUpsertRequestByDimension(upsertIds, dimension, namespace));
+        // Sometimes we see grpc failures when upserting so quickly after creating, so retry
+        assertWithRetry(() -> blockingStub.upsert(buildRequiredUpsertRequestByDimension(upsertIds, dimension,
+                namespace)), 1);
         dataPlaneConnection.close();
 
+        // Create collection from index
+        collection = createCollection(controlPlaneClient, collectionName, indexName, true);
+        assertEquals(collection.getName(), collectionName);
+        assertEquals(collection.getEnvironment(), environment);
+        assertEquals(collection.getStatus(), CollectionModel.StatusEnum.READY);
     }
 
     @AfterAll
     public static void cleanUp() throws InterruptedException {
         // wait for things to settle before cleanup...
         Thread.sleep(2500);
+
         // Clean up indexes
         for (String index : indexes) {
             controlPlaneClient.deleteIndex(index);
         }
-        // Clean up collections
-        for (String collection : collections) {
-            controlPlaneClient.deleteCollection(collection);
-        }
-        Thread.sleep(2500);
-    }
 
-    @Test
-    public void testIndexToCollectionHappyPath() throws InterruptedException {
-        String collectionName = RandomStringBuilder.build("collection-test", 8);
+        // Verify we can delete the collection
+        controlPlaneClient.deleteCollection(collectionName);
+        Thread.sleep(2500);
 
-        // Create collection from index
-        CollectionModel collection = createCollection(controlPlaneClient, collectionName, indexName, true);
-        collections.add(collectionName);
+        List<CollectionModel> collectionList = controlPlaneClient.listCollections().getCollections();
```

Comment:
Maybe this is overkill and we can just send the request to delete, but we used to validate the delete using list. I just moved that into the cleanup.